vendor/modules.txt

```diff
@@ -1097,7 +1097,7 @@ k8s.io/apiserver/plugin/pkg/audit/truncate
 k8s.io/apiserver/plugin/pkg/audit/webhook
 k8s.io/apiserver/plugin/pkg/authenticator/token/webhook
 k8s.io/apiserver/plugin/pkg/authorizer/webhook
-# k8s.io/autoscaler/vertical-pod-autoscaler v0.9.0 => k8s.io/autoscaler/vertical-pod-autoscaler v0.9.0
+# k8s.io/autoscaler/vertical-pod-autoscaler v0.9.0 => k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0
```

Comment:
I'm wondering why we see this? Is it because we should be updating `go.mod:50` rather to point to `0.10.0`?
```
-       k8s.io/autoscaler/vertical-pod-autoscaler v0.9.0
+       k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0
```

Reply:
I spend hours fighting with the go module system in this and previous PRs until I finally hit upon a state that works both locally and in CI. Please let's not spend more time on this.

Reply:
I tinkered with this a bit more. I was able to make the change suggested (use `0.10.0` directly in the `require` section), and even get rid of the corresponding replacement (`k8s.io/autoscaler/vertical-pod-autoscaler => k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0`) in the `replace` section. I was however not able to get rid of the second replacement (`k8s.io/autoscaler => k8s.io/autoscaler v0.0.0-20220125093630-6d1a1514af44`) - if I delete this line, `make revendor` locally still works, but CI complains with the following error:

```
github.com/gardener/gardener/cmd/gardener-extension-provider-local/app imports
	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1 loaded from k8s.io/autoscaler/vertical-pod-autoscaler@v0.10.0,
	but go 1.16 would fail to locate it:
	ambiguous import: found package k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1 in multiple modules:
	k8s.io/autoscaler v0.0.0-20190805135949-100e91ba756e (/home/prow/go/pkg/mod/k8s.io/autoscaler@v0.0.0-20190805135949-100e91ba756e/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1)
	k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0 (/home/prow/go/pkg/mod/k8s.io/autoscaler/vertical-pod-autoscaler@v0.10.0/pkg/apis/autoscaling.k8s.io/v1)
github.com/gardener/gardener/pkg/operation/botanist/component/etcd imports
	k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2 loaded from k8s.io/autoscaler/vertical-pod-autoscaler@v0.10.0,
	but go 1.16 would fail to locate it:
	ambiguous import: found package k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2 in multiple modules:
	k8s.io/autoscaler v0.0.0-20190805135949-100e91ba756e (/home/prow/go/pkg/mod/k8s.io/autoscaler@v0.0.0-20190805135949-100e91ba756e/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2)
	k8s.io/autoscaler/vertical-pod-autoscaler v0.10.0 (/home/prow/go/pkg/mod/k8s.io/autoscaler/vertical-pod-autoscaler@v0.10.0/pkg/apis/autoscaling.k8s.io/v1beta2)
```

Still this results in some simplification of `go.mod`, I hope it's ok to leave it like this.